File: 151002/Redkovskiy/StoryTeller/src/main/java/com/example/storyteller/model/Creator.java
package com.example.storyteller.model;

import jakarta.persistence.*;
import lombok.Data;

import java.util.HashSet;
import java.util.Set;

@Data
@Entity
@Table(name = "tbl_creator")
public class Creator {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 64)
    private String login;

    @Column(length = 128)
    private String password;

    @Column(length = 64)
    private String firstName;

    @Column(length = 64)
    private String lastName;

    @OneToMany(mappedBy = "creator", cascade = CascadeType.ALL)
    Set<Story> stories = new HashSet<>();
}

File: 151004/Akimenko/DC_task_02/dcPostgre/dcPostgre/Entities/DataModels/Creator.cs
using System.ComponentModel.DataAnnotations;
using static System.Net.Mime.MediaTypeNames;

namespace WebApplicationDC1.Entity.DataModel
{
    public class Creator(string login, string password, string firstName, string lastName) : Entity
    {
        [MinLength(2)]
        public string Login { get; set; } = login;
        [MinLength(8)]
        public string Password { get; set; } = password;
        [MinLength(2)]
        public string FirstName { get; set; } = firstName;
        [MinLength(2)]
        public string LastName { get; set; } = lastName;
        public ICollection<Story> Stories { get; set; } = [];

        public Creator(int id, string login, string password, string firstName, string lastName) :
            this(login, password, firstName, lastName)
        {
            Id = id;
        }

        public Creator() : this(string.Empty, string.Empty, string.Empty, string.Empty) { }
    }
}
